### PR TITLE
[CORL-2384] Upgrade to Node 18

### PR DESCRIPTION
**Pull request by nick-funk**
## What does this PR do?

Upgrades Coral to use Node 18 with npm 8.0.0.

## These changes will impact:

- [ ] commenters
- [ ] moderators
- [ ] admins
- [X] developers

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indices.

## How do I test this PR?

- install Node `v18.16.0` with npm `v8.0.0`
  - recommend using `nvm` for this
- run `sh scripts/clear-node_modules.sh`
  - this removes all existing node 14 module files that can cause conflicts
- run `sh npm-i.sh`
  - alternately, run `npm initialize.sh` if you've never run a build before or want to start fresh
- open a terminal in `client/` and run `npm run watch`
- open a terminal in `server/ and run `npm run watch`
- wait for the builds to finish
- navigate to http://localhost:8080 and test out the stream/admin
  - alternatively, do a server only build with `multi-site-test` if that's more to your taste
  
Extra testing:
- create an image using the `Dockerfile`
- spin up the image and see that it runs properly

## Where any tests migrated to React Testing Library?

No

## How do we deploy this PR?

No special considerations.

### CODE REVIEW

File: client/config/webpackDevServer.config.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ export default function ({
     publicPath,
     // WebpackDevServer is noisy by default so we emit custom message instead
     // by listening to the compiler events with `compiler.plugin` calls above.
-    quiet: true,
+    quiet: false,
```
Comment (kabeaty) [inline client/config/webpackDevServer.config.ts]: Just curious to know why we are setting `quiet` to `false` here now?

File: server/package.json [hunk 2]
```diff
@@ -368,10 +368,10 @@
     "terser-webpack-plugin": "^4.2.3",
```
Comment (kabeaty) [inline server/package.json:368]: Can we remove this now? And also the `terser-webpack-plugin.d.ts` types file too?

**Review by kabeaty — APPROVED** [review]
Looking great and works as expected! Just had two questions/comments about the changes.

File: client/config/webpackDevServer.config.ts (hunk 1 above)
Comment (nick-funk) [reply client/config/webpackDevServer.config.ts]: just me accidentally leaving this on from debugging watch on node 18, I've set it back to `true`

File: server/package.json (hunk 2 above)
Comment (nick-funk) [reply server/package.json:368]: removed!

Comment (kabeaty) [reply server/package.json:368]: yay thanks! looks like the types file was removed but not `terser-webpack-plugin` here too?

Comment (nick-funk) [reply server/package.json:368]: oh dang, I discarded at some point to try a cleaner approach to something or the other, must have lost removing this when I did that. I've fixed it up and pushed that change up now. Thanks for catching this!

**Review by tessalt — APPROVED** [review]

**Pull request merged**